legacy/src/main/java/org/opensearch/sql/legacy/executor/QueryActionElasticExecutor.java

```diff
@@ -45,6 +45,7 @@ public static SearchHits executeJoinSearchAction(
     ElasticJoinExecutor executor =
         ElasticJoinExecutor.createJoinExecutor(client, joinRequestBuilder);
     executor.run();
+    joinQueryAction.getPointInTimeHandler().deletePointInTime();
```

Comment:
Following previous comment, if we support both, here how we know if the JOIN is running with PIT or not?

Reply:
added condition to check